Resources/js/router_test.html

```diff
@@ -5,8 +5,8 @@
         <title>Router Test</title>
     </head>
     <body>
-        <script language="javascript" type="text/javascript" src="../../node_modules/google-closure-library/closure/goog/base.js"></script>
-        <script language="javascript" type="text/javascript" src="router.js"></script>
+        <script language="javascript" type="text/javascript" src="../node_modules/google-closure-library/closure/goog/base.js"></script>
+        <script language="javascript" type="text/javascript" src="../public/js/router.js"></script>
```

Comment:
Can the tests use the ES6 version? That way, you're always testing the non-compiled version, so compilation is not needed before you can run the tests.

Reply:
Attempting to do that was where Bruno's original PR came unstuck (see #257) because in order to do that requires rewriting the test suite as well as the code. Understandably people didn't think that was a good idea, and that PR has been stalled for over a year.

After a discussions with @weaverryan  and @stof at SymfonyCon, what I'm attempting to do here is to merge in the updated JS (to support webpack usage in SF4) without changing the test suite so that we can have a (high) degree of confidence that the new JS remains BC.

Reply:
All right, then I will accept this. However, the test script should invoke the compile step before running the test, to prevent one from testing without compiling.

Reply:
The docs should also be updated to reflect this (building before testing)

Reply:
Sure thing. Running `npm run test` will now perform a build first. Docs updated to explain that this will happen.